docs/book/src/component-config-tutorial/generate_docs.go

```diff
@@ -0,0 +1,196 @@
+webhook:
+  port: 9443
+leaderElection:
+  leaderElect: true
+  resourceName: 80807133.tutorial.kubebuilder.io
+clusterName: example-test`), 0600,
+	)
+	CheckError("fixing controller_manager_config", err)
+
+	// 2. add componentconfig
+	err = pluginutil.InsertCode(
+		filepath.Join(sp.ctx.Dir, "PROJECT"),
+		"More info: https://book.kubebuilder.io/reference/project-config.html",
+		"\ncomponentConfig: true",
+	)
+	CheckError("fixing PROJECT", err)
+
+	// 3. fix projectconfig_types.go
+	err = pluginutil.InsertCode(
+		filepath.Join(sp.ctx.Dir, "api/v2/projectconfig_types.go"),
+		"metav1 \"k8s.io/apimachinery/pkg/apis/meta/v1\"",
+		"\n  cfg \"sigs.k8s.io/controller-runtime/pkg/config/v1alpha1\"",
+	)
+	CheckError("fixing projectconfig_types", err)
+
+	err = pluginutil.InsertCode(
+		filepath.Join(sp.ctx.Dir, "api/v2/projectconfig_types.go"),
+		"Status ProjectConfigStatus `json:\"status,omitempty\"`",
+		//nolint:lll
+		"\n  // ControllerManagerConfigurationSpec returns the configurations for controllers\n  cfg.ControllerManagerConfigurationSpec `json:\",inline\"`\n\n  ClusterName string `json:\"clusterName,omitempty\"`",
+	)
+	CheckError("fixing projectconfig_types", err)
+
+	// 4. fix kustomization.yaml
+	err = pluginutil.InsertCode(
+		filepath.Join(sp.ctx.Dir, "config/default/kustomization.yaml"),
+		"manager_auth_proxy_patch.yaml",
+		//nolint:lll
+		"\n\n# Mount the controller config file for loading manager configurations\n# through a ComponentConfig type\n- manager_config_patch.yaml",
+	)
+	CheckError("fixing default/kustomization", err)
+
+	err = pluginutil.InsertCode(
+		filepath.Join(sp.ctx.Dir, "config/manager/kustomization.yaml"),
+		"- manager.yaml",
+		//nolint:lll
+		"\n\ngeneratorOptions:\n  disableNameSuffixHash: true\nconfigMapGenerator:\n- name: manager-config\n  files:\n  - controller_manager_config.yaml",
+	)
+	CheckError("fixing manager/kustomization", err)
+
+	// 5. fix manager_config_patch
+	err = pluginutil.InsertCode(
+		filepath.Join(sp.ctx.Dir, "config/default/manager_config_patch.yaml"),
+		"name: manager",
+		//nolint:lll
+		"\n        args:\n        - \"--config=controller_manager_config.yaml\"\n        volumeMounts:\n        - name: manager-config\n          mountPath: /controller_manager_config.yaml\n          subPath: controller_manager_config.yaml\n      volumes:\n      - name: manager-config\n        configMap:\n          name: manager-config",
+	)
+	CheckError("fixing manager_config_patch", err)
+
+	// 6. fix main
+	err = pluginutil.InsertCode(
+		filepath.Join(sp.ctx.Dir, "main.go"),
+		"ctrl.SetLogger(zap.New(zap.UseFlagOptions(&opts)))",
+		//nolint:lll
+		"\n\n  var err error\n  ctrlConfig := configv2.ProjectConfig{}\n  options := ctrl.Options{Scheme: scheme}\n  if configFile != \"\" {\n    options, err = options.AndFrom(ctrl.ConfigFile().AtPath(configFile).OfKind(&ctrlConfig))\n    if err != nil {\n      setupLog.Error(err, \"unable to load the config file\")\n      os.Exit(1)\n    }\n  }",
```

Comment:
It also should not be required because we are using the flag --component-config and that should do the scaffold accordingly. Can you try  --component-config=true just to see if the bug is we do have not a default true value? 
If you see that it is not working, then could you please raise an issue for us to fix as a follow? Up add a TODO here linking the issue so we are able to remove all after fixing it.

Reply:
Same as above. Fixed.